backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresModule.java

```diff
@@ -0,0 +1,129 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres;
+
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
+public interface PostgresModule {
+
+    static PostgresModule aggregateModules(PostgresModule... modules) {
+        return builder()
+            .modules(modules)
+            .build();
+    }
+
+    static PostgresModule aggregateModules(List<PostgresModule> modules) {
+        return builder()
+            .modules(modules)
+            .build();
+    }
+
+    PostgresModule EMPTY_MODULE = builder().build();
+
+    List<PostgresTable> tables();
+
+    List<PostgresIndex> tableIndexes();
+
+    class Impl implements PostgresModule {
+        private final List<PostgresTable> tables;
+        private final List<PostgresIndex> tableIndexes;
+
+        private Impl(List<PostgresTable> tables, List<PostgresIndex> tableIndexes) {
+            this.tables = tables;
+            this.tableIndexes = tableIndexes;
+        }
+
+        @Override
+        public List<PostgresTable> tables() {
+            return tables;
+        }
+
+        @Override
+        public List<PostgresIndex> tableIndexes() {
+            return tableIndexes;
+        }
+    }
+
+    class Builder {
+        private final ImmutableList.Builder<PostgresTable> tables;
+        private final ImmutableList.Builder<PostgresIndex> tableIndexes;
+
+        public Builder() {
+            tables = ImmutableList.builder();
+            tableIndexes = ImmutableList.builder();
+        }
+
+        public Builder table(PostgresTable... table) {
+            tables.add(table);
+            return this;
+        }
```

Comment:
Q: Should we change `table` -> `addTable`?

Today:
```
builder
  .table(tableA)
  .table(tableB)
```

My impression when use that code is tableB would override `tableA`, and only `tableB` is created.

Reply:
https://github.com/apache/james-project/pull/1770/commits/8089a3cb1adc20353817e213364a8ff45b9a8559